From 6032b78512685dc7d6cc3754d536823d428525c4 Mon Sep 17 00:00:00 2001
From: Felix Ernst <felixernst@zohomail.eu>
Date: Wed, 16 Apr 2025 13:50:05 +0000
Subject: [PATCH] Fix session restore

Since https://commits.kde.org/kxmlgui/8c9fb02a1d37672b26a03a9dd9e8675743deb269
KXmlGui only triggers a session restore for a window when the
session config contains a "NumberOfWindows" key with a value >= 1.
This commit adds such a key to the Dolphin session config, which
fixes the Dolphin main window restore for KXmlGui versions
containing that commit.

BUG: 502770


(cherry picked from commit c0bf226aa1cc02c9bffff1ec05e07a255d8d2f6e)

Co-authored-by: Felix Ernst <felixernst@zohomail.eu>
---
 src/dolphinmainwindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index 8a6c99ce57..391fe81a8c 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -757,6 +757,8 @@ void DolphinMainWindow::slotSaveSession()
         KConfig *config = KConfigGui::sessionConfig();
         saveGlobalProperties(config);
         savePropertiesInternal(config, 1);
+        KConfigGroup group = config->group(QStringLiteral("Number"));
+        group.writeEntry("NumberOfWindows", 1); // Makes session restore aware that there is a window to restore.
 
         auto future = QtConcurrent::run([config]() {
             config->sync();
-- 
GitLab

